ros2launch/ros2launch/api/api.py

```diff
@@ -154,11 +186,75 @@ def launch_a_launch_file(*, launch_file_path, launch_file_arguments, debug=False
             launch_arguments=parsed_launch_arguments,
         ),
     ])
+    if secure is not None:
+        # declare keystore so if it is a TemporaryDirectory, it gets cleaned up when out of scope
+        keystore, launch_description = _setup_security(  # noqa: F841
+            launch_description=launch_description,
+            keystore_path=secure,
+            create_keystore=create_keystore,
+        )
     launch_service.include_launch_description(launch_description)
     ret = launch_service.run()
```

Comment:
Does this block, then?

Reply:
Yes, this runs until all the actions finish, which means when all the nodes close